pallets/parachain-staking/src/lib.rs

```diff
@@ -2345,6 +2354,21 @@ pub mod pallet {
 			<AwardedPts<T>>::insert(now, author, score_plus_20);
 			<Points<T>>::mutate(now, |x| *x = x.saturating_add(20));
 		}
+
+		/// Cleans up historical staking information that is older than MaxOfflineRounds
+		/// by removing entries from the HadStake storage map.
+		fn cleanup_stake_info() {
+			let now = <Round<T>>::get().current;
+			let minimum_rounds_required = T::MaxOfflineRounds::get() + 1;
+
+			if now < minimum_rounds_required {
+				return;
+			}
+
+			let _ = <HadStake<T>>::iter_prefix(now - minimum_rounds_required)
+				.drain()
+				.next();
+		}
```

Comment:
If we properly estimate the weights in `on_initialize`, we could remove all entries from `WasInactive` for the round here.